### ISSUE TITLE
openPopup has wrong behaviour on CircleMarker

### ISSUE THREAD

**Issue by henus** [Leaflet/Leaflet#6910]
- [x] I've looked at the [documentation](http://leafletjs.com/reference.html) to make sure the behavior is documented and expected
- [x] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Cordova, Ionic, Angular, React…)
- [x] I've searched through the issues to make sure it's not yet reported

**Steps to reproduce**
- Made second click on CircleMarker with popup.

**Expected behavior**
- Second click on CircleMarker close popup.

**Current behavior**
- Second click on CircleMarker open the popup (at the new location).

**Environment**
- Leaflet version: 1.6.0
- Browser (with version): Chrome (76) 
- OS/Platform (with version): Windows 10

**Additional context**
Method openPopup over layers that inherits from Path just open the popup at the new location.

CircleMarker inherits from Path, but its appearance on map and behaviour similar to simple Marker (radius of CircleMarker doesn`t changed on map zoom).

**Minimal example reproducing the issue**
https://jsfiddle.net/uf7k9a6v/1/

- [x] this example is as simple as possible
- [x] this example does not rely on any third party code

**Comment by IvanSanchez** [comment]
No, this is behaviour as expected. `L.CircleMarker`s behave as `L.Path`s, just like `L.Polygon`s do. See the example at https://leafletjs.com/examples/map-panes/ : clicking twice on a country will re-display the same pop-up. That's the expected behaviour.

**Comment by henus** [comment]
It`s a good example with `L.Polygon`, but wrong conclusion.
- `L.Polygon` changes its size on map zoom and determines the area on the map, so re-display the same pop-up on new click coordinates is justified.
- `L.CircleMarker` does not change its size on map zoom, it determines a point on the map, so re-display the same pop-up on new click coordinates is unjustified.

Behaviour of `L.CircleMarker` should be similar to `L.Marker` (second click on CircleMarker close popup)

**Comment by johnd0e** [comment]
https://github.com/Leaflet/Leaflet/blob/0f904a515879fcd08f69b7f51799ee7f18f23fd8/src/layer/Popup.js#L487-L500

Has some common with: #6656.

Instead of hardcoding such condition - better introduce some official property for `Layer`.
